### DISCUSSION TITLE
In the scenario of TCP half-closed connection, after the client calls shutdownOutput, why does the server continuously receive read_ready notifications from selector  in the close_wait state?

### DISCUSSION THREAD

**Discussion by huibinliupush** [netty/netty#11962] [Q&A]
## question
![11](https://user-images.githubusercontent.com/5794695/147722976-9f4d409e-2b45-4b7e-8b8b-6dacf26e5d77.png)

In the scenario of TCP half-closed connection, after the client calls shutdownOutput, **why does the server continuously receive read_ready notifications from selector  in the close_wait state?**

## my analysis

i know this problem has bee solved in [this PR](https://github.com/netty/netty/commit/ed0668384b393c3502c2136e3cc412a5c8c9056e)。

![image](https://user-images.githubusercontent.com/5794695/147723109-fbc61f2e-b627-4de5-8531-8f854a11d968.png)

I want to know that since we have read` EOF` and the `doReadBytes` method returns` -1,` in this case why the selector will keep notifying the read event to be active until the server calls the `close` method to end the `close_wait` state

**Comment by littleyg** [comment]
Actually, Java NIO Selector does this NIO read spin event loop spin when half closed.

**Comment by normanmaurer** [comment]
I think this was fixed by https://github.com/netty/netty/pull/11982 ?
